spacy_llm/tasks/entity_linker/util.py

```diff
@@ -0,0 +1,190 @@
+            )
+        example.reference.ents = [
+            ent for ent in example.reference.ents if ent.kb_id != 0
+        ]
+        if len(example.reference.ents) == 0:
+            return None
+
+        # Assemble example.
+        mentions = [ent.text for ent in example.reference.ents]
+        # Fetch candidates. If true entity not among candidates: fetch description separately and add manually.
+        cands_ents, solutions = task.fetch_entity_info(example.reference)
+        # If we are to use available docs as examples, they have to have KB IDs set and hence available solutions.
+        assert all([sol is not None for sol in solutions])
+
+        return ELExample(
+            text=EntityLinkerTask.highlight_ents_in_text(example.reference),
+            mentions=mentions,
+            entity_descriptions=[
+                [ent.description for ent in ents] for ents in cands_ents
+            ],
+            entity_ids=[[ent.id for ent in ents] for ents in cands_ents],
+            solutions=solutions,
+            reasons=[""] * len(mentions),
+        )
+
+
+def score(examples: Iterable[Example], **kwargs) -> Dict[str, Any]:
+    """Score entity linking accuracy in examples.
+    examples (Iterable[Example]): Examples to score.
+    RETURNS (Dict[str, Any]): Dict with metric name -> score.
+    """
+    return Scorer.score_links(examples, negative_labels=[EntityLinker.NIL])
+
+
+def ent_desc_reader_csv(path: Union[Path, str]) -> Dict[str, str]:
+    """Instantiates entity description reader with two columns: ID and description.
+    path (Union[Path, str]): File path.
+    RETURNS (Dict[str, str]): Dict with ID -> description.
+    """
+    with open(path) as csvfile:
+        descs: Dict[str, str] = {}
+        for row in csv.reader(csvfile, quoting=csv.QUOTE_ALL, delimiter=";"):
+            if len(row) != 2:
+                continue
+            descs[row[0]] = row[1]
+
+        if len(descs) == 0:
+            raise ValueError(
+                "Format of CSV file with entity descriptions is wrong. CSV has to be formatted as "
+                "semicolon-delimited CSV with two columns. The first columns has to contain the entity"
+                " ID, the second the entity description."
+            )
+
+    return descs
+
+
+@dataclasses.dataclass
+class BaseInMemoryLookupKBLoader:
+    path: Union[str, Path]
+    """Path to artefact containing knowledge base data."""
+
+    def __post_init__(self):
+        if isinstance(self.path, str):
+            self.path = Path(self.path)
+
+    @abc.abstractmethod
+    def __call__(self, vocab: Vocab) -> Tuple[InMemoryLookupKB, DescFormat]:
+        """Loads KB instance.
+        vocab (Vocab): Vocab instance of executing pipeline.
+        RETURNS (Tuple[InMemoryLookupKB, DescFormat]): Loaded/generated KB instance; descriptions for entities.
+        """
+        ...
+
+
+@dataclasses.dataclass
+class KBSerializedLoader(BaseInMemoryLookupKBLoader):
```

Comment:
If we call the other thing `KBFileLoader`, perhaps this could be `KBObjectLoader`? To me that makes sense (this is a serialized object, the other a plain text file) but I'm not sure whether this will be intuitive naming for others. Happy to discuss more.